tests/sanity/tests/model/contacts/contract-page.ts

```diff
@@ -0,0 +1,244 @@
+  readonly openDeleteButton = (): Locator => this.page.locator('button:has-text("Delete")')
+  readonly newApplicationDescription = (): Locator => this.page.getByRole('paragraph')
+  readonly newApplicationAssignRectruiter = (): Locator => this.page.getByRole('button', { name: 'Assigned recruiter' })
+  readonly newApplicationChooseRecruiter = (recruiter: string): Locator => this.page.getByRole('button', { name: recruiter })
+  readonly newApplicationInterview = (Interview: string): Locator => this.page.getByRole('button', { name: Interview })
+  readonly newApplicationStartDate = (): Locator => this.page.getByRole('button', { name: 'Start date' })
+  readonly newApplicationStartInADay = (): Locator => this.page.getByText('in a day')
+  readonly newApplicationCreate = (): Locator => this.page.getByRole('button', { name: 'Create' })
+  readonly personName = (person: string): Locator => this.page.locator(`text=${person}`)
+  readonly personTable = (): Locator => this.page.locator('.antiTable-body__row')
+  readonly personMarina = (): Locator => this.page.getByRole('link', { name: 'MM M. Marina' })
+  readonly comapnyTab = (): Locator => this.page.locator('.antiNav-element:has-text("Company")')
+  readonly addCompany = (): Locator => this.page.locator('button:has-text("Company")')
+  readonly companyName = (): Locator => this.page.locator('[placeholder="Company name"]')
+  readonly companyCreateButton = (): Locator => this.page.locator('button:has-text("Create")')
+  readonly companyByName = (company: string): Locator => this.page.locator(`text=${company}`)
+  readonly addMember = (): Locator => this.page.locator('[id="contact:string:AddMember"]')
+  readonly selectMember = (): Locator => this.page.getByRole('cell', { name: 'CR Chen Rosamund' }).getByRole('link')
+  readonly openNewMember = (member: string): Locator => this.page.locator(`.card a:has-text("${member}")`)
+  readonly newMemberAdded = (): Locator => this.page.getByText('New Members: Chen Rosamund')
+  readonly stateApplication = (role: string): Locator => this.page.getByRole('cell', { name: role })
+  readonly commentApplication = (): Locator => this.page.getByRole('button', { name: '1', exact: true })
+  readonly commentDescription = (): Locator => this.page.getByText('Test Application')
+
+  // ACTIONS
+
+  async addNewApplication (description: string, recruiter: string): Promise<void> {
+    await this.newApplicationDescription().click()
+    await this.newApplicationDescription().fill(description)
+    await this.newApplicationAssignRectruiter().click()
+    await this.newApplicationChooseRecruiter(recruiter).click()
+    await this.newApplicationStartDate().click()
+    await this.newApplicationStartInADay().click()
+    await this.newApplicationCreate().click()
+  }
+
+  async clickAppContact (): Promise<void> {
+    await this.appContact().click()
+  }
+
+  async clickEmployeeNavElement (Employee: string): Promise<void> {
```

Comment:
Not sure we need all of this methods, why no use locator we already have?

Reply:
Yes I noticed that you have locators separated inside your app in a smaller chunks and that is a great way to do it, honestly, I  did not want to use or change much of it, this was more of adding my own approach. As for the methods, the best practice is to define the locators, define the actions, and assert using those locators in the function and then call those functions inside the test, so basically `page. locator("#something).click()` is not the best approach. 
It usually is one action per locator but it can vary and we can create functions for example called logInUser where it would look like 

```
loginUser(username: stirng, password: string) {
   await username.fill(username)
   await pasword.fill(password)
   await login.click()
}
```

Reply:
Also I am up for using functions for doing multiple actions", but for the sake of this PR and my limited knowledge of app I went with the conventional mostly on one action per function. Unless I was certain that the same approach cane be reused with multiple actions inside the function like:

```
  async addNewApplication (description: string, recruiter: string): Promise<void> {
    await this.newApplicationDescription().click()
    await this.newApplicationDescription().fill(description)
    await this.newApplicationAssignRectruiter().click()
    await this.newApplicationChooseRecruiter(recruiter).click()
    await this.newApplicationStartDate().click()
    await this.newApplicationStartInADay().click()
    await this.newApplicationCreate().click()
  }
```

```
  async kickEmployee (first: string, last: string): Promise<void> {
    await this.employeeEntry(first, last).hover()
    await this.employeeEntry(first, last).click({ button: 'right' })
    await this.kickEmployeeOption().click()
    await this.okButton().click()
  }
```

Reply:
Makes sense to me, the test case code looks much cleaner now

Reply:
thank you @aonnikov